api/v1beta1/network_disruption.go

```diff
@@ -150,23 +172,118 @@ type NetworkDisruptionCloudServiceSpec struct {
+		retErr = multierror.Append(retErr, fmt.Errorf("the Method specification at the HTTP network disruption level is deprecated; use Methods HTTP field instead"))
+	}
+
+	if len(s.Paths) > MaxNetworkPaths {
+		retErr = multierror.Append(retErr, fmt.Errorf(HTTPPathsFilterErrorPrefix+"the number of paths must not be greater than %d; Number of paths: %d", MaxNetworkPaths, len(s.Paths)))
+	} else if len(s.Paths) > 0 {
+		visitedPaths := make(map[HTTPPath]struct {
+			count int
+		})
+
+		isMultiplePath := false
+		if len(s.Paths) > 1 {
+			isMultiplePath = true
 		}
 
-		if regexp.MustCompile(`\s`).MatchString(s.Path) {
-			return fmt.Errorf("the path specification at the network disruption level is not valid; should not contains spaces")
+		for _, path := range s.Paths {
+			visitedPath, isVisited := visitedPaths[path]
+			if isVisited {
+				visitedPath.count++
+				visitedPaths[path] = visitedPath
+				continue
+			}
+
+			visitedPath.count++
+			visitedPaths[path] = visitedPath
+
+			if isMultiplePath && path == DefaultHTTPPathFilter {
+				retErr = multierror.Append(retErr, fmt.Errorf(HTTPPathsFilterErrorPrefix+"no needs to define other paths if the / path is defined because it already catches all paths"))
+			}
+
+			if err := path.validate(); err != nil {
+				retErr = multierror.Append(retErr, err)
+			}
 		}
 
-		if string(s.Path[0]) != DefaultHTTPPathFilter {
-			return fmt.Errorf("the path specification at the network disruption level is not valid; should start with a /")
+		for path, visitedPath := range visitedPaths {
+			if visitedPath.count > 1 {
+				retErr = multierror.Append(retErr, fmt.Errorf(HTTPPathsFilterErrorPrefix+"should not contain duplicated paths. Count: %d; Path: %s", visitedPath.count, path))
+				delete(visitedPaths, path)
+			}
 		}
 	}
 
-	return nil
+	if len(s.Methods) > MaxNetworkMethods {
+		retErr = multierror.Append(retErr, fmt.Errorf(HTTPMethodsFilterErrorPrefix+"the number of methods must not be greater than %d; Number of methods: %d", MaxNetworkMethods, len(s.Methods)))
+	} else if len(s.Methods) > 0 {
+		visitedMethods := make(map[string]struct {
+			count int
+		})
+
+		for _, method := range s.Methods {
+			if _, ok := allowedHTTPMethods[method]; !ok {
+				err := fmt.Errorf(HTTPMethodsFilterErrorPrefix+"should be a GET, DELETE, POST, PUT, HEAD, PATCH, CONNECT, OPTIONS or TRACE. Invalid value: %s", method)
+				retErr = multierror.Append(retErr, err)
+				continue
+			}
+
+			visitedMethod, isVisited := visitedMethods[method]
+			if isVisited {
+				visitedMethod.count++
+				visitedMethods[method] = visitedMethod
+				continue
+			}
+
+			visitedMethod.count++
+			visitedMethods[method] = visitedMethod
+		}
+
+		for method, visitedMethod := range visitedMethods {
+			if visitedMethod.count > 1 {
+				retErr = multierror.Append(retErr, fmt.Errorf(HTTPMethodsFilterErrorPrefix+"should not contain duplicated methods. Count: %d; Method: %s", visitedMethod.count, method))
+				delete(visitedMethods, method)
```

Comment:
```suggestion
```

I don't think we should mutate this while we're iterating through it. It also shouldn't have an effect on our behavior regardless? We'll only list one error per method

Reply:
That's true! I 👍